src/plugins/mission/mission_impl.cpp

```diff
@@ -581,13 +584,19 @@ std::pair<Mission::Result, Mission::MissionPlan> MissionImpl::convert_to_result_
 
             } else if (int_item.command == MAV_CMD_DO_CHANGE_SPEED) {
                 if (int(int_item.param1) == 1 && int_item.param3 < 0 && int(int_item.param4) == 0) {
-                    new_mission_item.speed_m_s = int_item.param2;
+                    cached_speed = int_item.param2;
+                    new_mission_item.speed_m_s = cached_speed;
                 } else {
                     LogErr() << "Mission item DO_CHANGE_SPEED params unsupported";
                     result_pair.first = Mission::Result::Unsupported;
                 }
 
             } else if (int_item.command == MAV_CMD_NAV_LOITER_TIME) {
+                // MAVSDK doesn't use LOITER_TIME anymore, but it is possible a mission still uses
+                // it
+                new_mission_item.loiter_time_s = int_item.param1;
```

Comment:
We should either support minutes, hours, etc. as well, or at least print an error if they are set and we are ignoring them.